src/SharedUtilities/utils.jl

```diff
@@ -17,3 +19,156 @@ end
 Helper function used in computing tendencies of vertical diffusion terms.
 """
 to_scalar_coefs(vector_coefs) = map(vector_coef -> vector_coef.u₃, vector_coefs)
+
+"""
+   dss!(Y::ClimaCore.Fields.FieldVector, p::ClimaCore.Fields.FieldVector, T::FT)
+
+Computes the weighted direct stiffness summation and updates `Y` in place.
+In the case of a column domain, no dss operations are performed.
+"""
+function dss!(
+    Y::ClimaCore.Fields.FieldVector,
+    _::ClimaCore.Fields.FieldVector,
+    _::FT,
+) where {FT}
+    for key in propertynames(Y)
+        property = getproperty(Y, key)
+        dss_helper!(property, axes(property))
+    end
+end
+
+"""
+    dss_helper!(field_vec::ClimaCore.Fields.FieldVector, _)
+
+Method of `dss_helper!` which unpacks properties of Y when on a
+domain that is 2-dimensional in the horizontal.
+
+The assumption is that Y contains FieldVectors which themselves contain either
+FieldVectors or Fields, and that the final unpacked variable is a Field.
+This method is invoked when the current property itself contains additional
+property(ies).
+"""
+function dss_helper!(field_vec::ClimaCore.Fields.FieldVector, _)
```

Comment:
unit test to make sure recursion works?

Reply:
I added a test for dss which also checks that the recursion works